### PR TITLE
[docs] Add file watcher to performance section

### PR DESCRIPTION
**Pull request by ericdallo**

### CODE REVIEW

File: docs/page/performance.md [hunk 1]
```diff
@@ -41,6 +41,16 @@ Benchmarks show that Emacs 27 is `~15 times` faster than Emacs when using Elisp
 (setq lsp-idle-delay 0.500)
 ```
 
+## Ignore watch folders/files
+
+By default `lsp-mode` tries to watch all files and folders of the project ignoring the regexp from `lsp-file-watch-ignored`. If you don't want some file or folder
```
Comment (yyoncho) [inline docs/page/performance.md:46]: This will happen only if the server wants to watch files in the project.

Comment (ericdallo) [reply docs/page/performance.md:46]: If the server has the capability you mean? [this](https://microsoft.github.io/language-server-protocol/specification#workspace_didChangeWatchedFiles) capability?

Comment (yyoncho) [reply docs/page/performance.md:46]: Not exactly. The server might have the capability but we start watching the folder after server requests that.

Comment (ericdallo) [reply docs/page/performance.md:46]: I see, I'll update it

Comment (ericdallo) [reply docs/page/performance.md:46]: BTW I found nothing about server requesting that to the client in the [spec](https://microsoft.github.io/language-server-protocol/specifications/specification-3-16/)

Comment (yyoncho) [reply docs/page/performance.md:46]: What I meant is that not about server having the capability but about server sending `DidChangeWatchedFilesRegistrationOptions`

Comment (ericdallo) [reply docs/page/performance.md:46]: Oh got it!

**Pull request merged**